provider/sdk/dkubefs/online_drivers/remote_driver.py

```diff
@@ -0,0 +1,279 @@
+        #     _query = f"""
+        #         select entity_key, feature_name, value, event_ts from
+        #         {_table_name(project, table)} where
+        #         entity_key = %s
+        #     """
+        #     with connect(**self.connect_args) as conn:
+        #         with conn.cursor(buffered=True) as cursor:
+        #             cursor.execute(_query, (entity_key_bin,))
+        #             res, res_ts = dict(), None
+        #             for _, _feature_name, _value, _ts in cursor.fetchall():
+        #                 val = ValueProto()
+        #                 val.ParseFromString(_value)
+        #                 res[_feature_name] = val
+        #                 res_ts = _ts
+
+        #             if not res:
+        #                 result.append((None, None))
+        #             else:
+        #                 result.append((res_ts, res))
+        # return result
+        pass
+
+    def update(
+        self,
+        config: RepoConfig,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
+        entities_to_delete: Sequence[Entity],
+        entities_to_keep: Sequence[Entity],
+        partial: bool,
+    ) -> None:
+        project = config.project
+        (
+            keep_tables_names,
+            delete_tables_names,
+            keep_entities_names,
+            delete_entities_names
+        ) = preprocess_infra_tables(
+            project=project,
+            tables_to_delete=tables_to_delete,
+            tables_to_keep=tables_to_keep,
+            entities_to_delete=entities_to_delete,
+            entities_to_keep=entities_to_keep
+            )
+        tables_data = dict(
+            project=project,
+            tables_to_delete=delete_tables_names,
+            tables_to_keep=keep_tables_names,
+            entities_to_keep=keep_entities_names,
+            entities_to_delete=delete_entities_names,
+            user=self.user,
+            offline_dataset=self.offline_dataset
+            )
+        self.online_server_client.post("api/v1/infra_update",
+                                       data=tables_data,
+                                       headers=self.online_server_host)
+
+    def teardown(
+        self,
+        config: RepoConfig,
+        tables: Sequence[Union[FeatureTable, FeatureView]],
+        entities: Sequence[Entity],
+    ) -> None:
+        # teardown_infra should remove all deployed infrastructure
+        # Replace the code below in order to define your own custom teardown
+        # operations
+        project = config.project
+        tables_to_teardown = preprocess_teardown_tables(project, tables)
+        teardown_data = {
+            "project": project,
+            "tables": tables_to_teardown,
+            "user": self.user,
+            "offline_dataset": self.offline_dataset
+        }
+        self.online_server_client.delete("api/v1/teardown",
```

Comment:
Can we call dkube sdk whenever we need to make API calls?

Reply:
This call goes for knative service independent of controller. This is the reason, not part of Dkube SDK.